sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/AadRestTemplateCreator.java

```diff
@@ -27,25 +24,14 @@ public static RestTemplate createRestTemplate(RestTemplateBuilder builder) {
     }
 
     public static RestTemplate createOAuth2ErrorResponseHandledRestTemplate(RestTemplateBuilder builder) {
-        RestTemplate restTemplate = createRestTemplate(builder);
-        restTemplate.setErrorHandler(new OAuth2ErrorResponseErrorHandler());
-        return restTemplate;
+        builder = builder.errorHandler(new OAuth2ErrorResponseErrorHandler());
+        return createRestTemplate(builder);
     }
 
     public static RestTemplate createOAuth2AccessTokenResponseClientRestTemplate(RestTemplateBuilder builder) {
-        RestTemplate restTemplate = createOAuth2ErrorResponseHandledRestTemplate(builder);
-        List<HttpMessageConverter<?>> converters = restTemplate.getMessageConverters();
-        if (notContainsElementOfType(converters, FormHttpMessageConverter.class)) {
-            converters.add(new FormHttpMessageConverter());
-        }
-        if (notContainsElementOfType(converters, OAuth2AccessTokenResponseHttpMessageConverter.class)) {
-            converters.add(new OAuth2AccessTokenResponseHttpMessageConverter());
-        }
-        return restTemplate;
-    }
-
-    private static boolean notContainsElementOfType(List<?> list, Class<?> clazz) {
-        return list.stream().noneMatch(item -> item.getClass().equals(clazz));
+        builder = builder.messageConverters(
```

Comment:
Shall we set the message converters and error handler based on the rest template instance, not the rest template builder?

Reply:
It's the same. The `build.xxx` will create a new builder instance instead of modifying current instance.

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/31213#discussion_r984260591